pkg/frontend/session.go

```diff
@@ -1677,20 +1676,67 @@ func (tcc *TxnCompilerContext) Resolve(dbName string, tableName string) (*plan2.
 					Partition: p,
 				},
 			})
-		} else if indexDef, ok := def.(*engine.ComputeIndexDef); ok {
-			fields := make([]*plan.Field, len(indexDef.Fields))
-			for i := range indexDef.Fields {
+		} else if indexDef, ok := def.(*engine.UniqueIndexDef); ok {
```

Comment:
Why do we want to do runtime type cast?    can all these indexdef implement a common, well defined interface?